cranelift/codegen/src/isa/riscv64/lower.isle

```diff
@@ -612,14 +612,17 @@
   (gen_stack_addr ss offset))
 
 ;;;;;  Rules for `is_null`;;;;;;;;;
-(rule
-  (lower (is_null v))
-  (gen_reference_check (ReferenceCheckOP.IsNull) v))
+
+;; Null references are represented by the constant value `0`.
+(rule (lower (is_null v))
+  (seqz v))
 
 ;;;;;  Rules for `is_invalid`;;;;;;;;;
-(rule
-  (lower (is_invalid v))
-  (gen_reference_check (ReferenceCheckOP.IsInvalid) v))
+
+;; Invalid references are represented by the constant value `-1`.
+(rule (lower (is_invalid v))
+  (let ((v Reg (alu_rr_imm12 (AluOPRRI.Addi) v (imm12_const 1))))
+    (seqz v)))
```

Comment:
I'd be inclined to not shadow `v` here with a new definition. It's not a bug or anything, but I had to read a little more carefully to see that the seqz consumes the redefinition, not the original input.

I think you could inline it, actually, and maybe that's more clear? I don't know, I haven't developed strong opinions about style in ISLE source yet.
```suggestion
  (seqz (alu_rr_imm12 (AluOPRRI.Addi) v (imm12_const 1))))
```

Reply:
>I think you could inline it, actually, and maybe that's more clear? I don't know, I haven't developed strong opinions about style in ISLE source yet.

I haven't either, but it would be nice to have some sort of consensus among backends!

One of the things I noticed while working in the RISC-V backend is that we don't seem to define too many instructions helpers. We always do something like `(alu_rrr (opcode) ...)`, I'm personally not a big fan of this, but I'm not sure how other people feel about this stuff.

Also maybe we could get some sort of lisp formatter that would work for folks? Styles are different across all backends, It'd be nice to standardize on *something*. 

Maybe that is something for https://github.com/bytecodealliance/wasmtime/issues/5752, I'm not sure how far LSP goes.